src/clinical/api/clinical-api.ts

```diff
@@ -186,15 +183,39 @@ class ClinicalController {
 
 		const donorEntityData = await service.getDonorEntityData(donorIds);
 
+		const entityRecords = donorEntityData[0].records;
+
+		// File Table can request multiple programs
+		const donorPrograms = entityRecords
+			.map((record) => record.program_id as string)
+			.filter((programId, index, array) => programId && array.indexOf(programId) === index);
+
 		const date = currentDateFormatted();
 		const fileName = `filename=Donor_Clinical_Data_${date}.zip`;
+
+		const exceptions = await Promise.all(
+			donorPrograms.map(async (programShortName) => {
+				const programDonors = entityRecords
+					.filter((record) => record.donor_id && record.program_id === programShortName)
+					.map((record) => record.donor_Id as number);
+
+				const programExceptions =
+					(await getExceptionManifestRecords(programShortName, {
+						donorIds: programDonors,
+						submitterDonorIds: [],
+					})) || [];
+
+				return programExceptions;
+			}),
+		).then((exceptionsArray) => exceptionsArray.flat());
```

Comment:
I'm going to accept this PR without these changes, but this is some pretty heavily nested code thats hard for me to parse. Imagine it reorganized like:

```ts
const getEntityExceptionsByProgramName = async (programShortName: string) => {
	const programDonors = entityRecords
		.filter((record) => record.donor_id && record.program_id === programShortName)
		.map((record) => record.donor_Id as number);
	const programExceptions =
		(await getExceptionManifestRecords(programShortName, {
			donorIds: programDonors,
			submitterDonorIds: [],
		})) || [];

	return programExceptions;
}

const programExceptionPromises = donorPrograms.map(getProgramExceptionsForDonors);

const exceptionsArray = await Promise.all(programExceptionPromises);
const exceptions = exceptionsArray.flat();

```

Reply:
Yes async maps get very ugly ... I improved readability but used a different approach w/ a `for ... of` loop